### ISSUE TITLE
Regularisierung

### ISSUE THREAD

**Issue by inkrement** [tridelt/PyTorchWord2Vec#1]
Eine Verbesserung wäre eventuell Regularisierung. Bekannt ist hier vor allem die l2 Regularisierung die du direkt bei dem [Embedding](https://github.com/tridelt/PyTorchWord2Vec/blob/aef3532ea6325b37bde76b75f5bbcd946bfe0f7b/model.py#L11) angeben könntest (soweit ich das im Kopf habe). Da gibt es zB einen max_norm Parameter der dir hilfreich sein könnte (https://pytorch.org/docs/stable/nn.html#embedding).

Mehr Infos über Regularisierung findest du zB hier: [https://arxiv.org/pdf/1508.03721.pdf](https://arxiv.org/pdf/1508.03721.pdf)

**Comment by tridelt** [comment]
Welchen Wert soll ich für max_norm verwenden? Ich gedenke zu verstehen, dass man mittels der Normalisierung besonders grosse Werte, welche sich als Phänomen des unsupervised Learnings bilden, bestrafen möchte, aber wie finde ich einen Wert der gut zu meinem Modell passt?

    Die Werte meines Embeddings ausprinten und dann Trial and Error?

**Comment by inkrement** [comment]
Da gibt es mehrere Möglichkeiten: 1) Recherchieren und "best practice Werte nutzen" (habe zB eine Publikation genannt. Die hier https://github.com/RasaHQ/rasa_nlu/blob/master/rasa_nlu/classifiers/embedding_intent_classifier.py haben zB 0.002 gewählt. 2) oder du siehst es als Hyperparameter an und versuchst einen optimalen Wert zu finden. Da deine Methode recht lange braucht, würde ich fast davon abraten oder auf einem Subset optimieren.

LG

**Comment by inkrement** [comment]
Hat sich glaub ich mittlerweile erledigt
